Assets/MixedRealityToolkit/Providers/Hands/BaseHand.cs

```diff
@@ -9,7 +9,7 @@ namespace Microsoft.MixedReality.Toolkit.Input
     public abstract class BaseHand : BaseController, IMixedRealityHand
     {
         // Hand ray
-        protected HandRay HandRay { get; } = new HandRay();
+        protected IHandRay HandRay { get; } = new HandRay();
```

Comment:
Before we merge this, what are opinions of @wiwei and @keveleigh on making this field virtual?

Reply:
Seems fine to me as it would remind people to provide an implementation. Currently if somebody extends BaseHand and forgets to specify HandRay, the default value will be used.

Reply:
My envisioned next step would be to move the default type into a profile, though making this virtual would allow specific hand types to override even the profile-set type, which I could see being desirable.

Reply:
Alright just updated the code with this change.

Reply:
I would prefer to not add more things into profiles as it's quite hard to discover these by reading the code / docs.

Reply:
I'm not sure of another way to make this configurable without touching code (for example, in precompiled MRTK) without adding it to, say, a type dropdown in the hand tracking profile. Do you have other ideas?

Reply:
Since it's more than a change in visualization (in which case, the visualization prefabs could be usable), but it's a change in the way the pointer pose is calculated.